zigzag-porep.md

```diff
@@ -196,22 +197,23 @@ func DeriveChallenges(challenges LayerChallenges, layer uint, leaves uint,
 
 ### Layer Replication
 
+TODO: Define `Graph`. We need to decide if this is an object we'll explicitly define or if its properties (e.g., `GRAPH_SIZE`) are just part of the replication parameters and all the functions just refer to the _same_ graphs being manipulated across the entire replication process. (At the moment I've avoided defining a `Graph` structure as in other specs I didn't see any object methods, just standalone functions.)
+
```

Comment:
I don't think `Graph` needs to be a first-class part of the spec. It's an implementation detail of how the problem is decomposed in `rust-fil-proofs` as a proof toolkit.

However, the cleanest way to specify based on what we have might still involve a small amount of state to track the direction of the current layer. If bundling that and any other convenience state into a `Graph` struct makes sense, that seems reasonable, at least for now.